### PR TITLE
RAD-382 Create RadiologyReportServiceTest

### PR DESCRIPTION
**Pull request by manuela-grindei**
## Description

Created unit tests for RadiologyReportServiceTest and removed unnecessary integration tests.
## Related Issue

see https://issues.openmrs.org/browse/RAD-382

## Checklist:

- [x] My pull request only contains one single commit.
- [x] My pull request is based on the latest master branch
  `git pull --rebase upstream master`.
- [x] I ran `mvn clean package` right before creating this pull request and
  added all formatting changes to my commit.
- [x] My code follows the code style of this project.
- [x] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

**Comment by dkayiwa** [comment]
@teleivo do you think this is fit to merge?

**Comment by dkayiwa** [comment]
@ivange94 do you think you can review this? :)

**Comment by ivange94** [comment]
That was suppose to be a comment instead of suggestion of changes.  :D

**Comment by manuela-grindei** [comment]
@ivange94 Hello, thank you for reviewing this!
 I skipped that one on purpose, as the class under test is not doing actual validation, but calling the validator. In these conditions, I thought it would be best to leave it as an integration. I could add a unit test using doThrow(Exception), but I'm not sure it would be that valuable.

**Comment by ivange94** [comment]
@manuela-grindei I didn't get your point as to why that should be an integration test. BTW the issue says "Extract tests that do not need an ApplicationContext and DB into a unit test class RadiologyReportServiceTest...". And that tests falls in that category.

**Comment by ivange94** [comment]
@manuela-grindei yes you are right. Just forget about my previous comment(I deleted it). Just stepped through the code. I think this is a testing behavior vs testing implementation problem. To my knowledge we test for behavior, thats why we use mocks and in this case what that test is doing is testing how that method will behave when an invalid report is passed. Its not testing who is actually doing the validation(which I think is testing the implementation). So I still think you should extract that test method out of the integration test. @dkayiwa what do you think?

**Comment by dkayiwa** [comment]
@teleivo now that you are back, do you think we can merge this? :)

**Comment by dkayiwa** [comment]
@teleivo should i help you merge this? :)

File: api/src/test/java/org/openmrs/module/radiology/report/RadiologyReportServiceTest.java [hunk 1]
```diff
@@ -0,0 +1,346 @@
...
+import org.openmrs.api.APIException;
+import org.openmrs.module.radiology.order.RadiologyOrder;
+import org.openmrs.test.BaseContextMockTest;
+
+import static org.mockito.Mockito.when;
+
+/**
+ *  Tests {@link RadiologyReportService}
+ */
+@RunWith(MockitoJUnitRunner.class)
```
Comment (manuela-grindei) [inline api/src/test/java/org/openmrs/module/radiology/report/RadiologyReportServiceTest.java]: Good point, removed it!

**Comment by dkayiwa** [comment]
@teleivo what do you think of this? 😊

**Comment by teleivo** [comment]
@manuela-grindei thank you very much and sorry for taking so long! 😉 @dkayiwa

**Pull request merged**